### ISSUE TITLE
Conditional render on model update

### ISSUE THREAD

**Issue by rsaccon** [David-OConnor/seed#14]
If I understand properly, each time the `update` function changes the model, a `render`  gets triggered. 

This is a problem with my recent experiment to add animations based on a `requestAnimationFrame` loop to control spring-like animations. Whether the animations should run or not, depends on user interactions, e.g.  if a button click triggers to open a modal then the animation fades or slides the modal in. I want to store the animation metadata (which changes during the lifetime of an animation) in a hashmap in the model. But updating that metadata triggers a 'render'. I don't want that, because the animations work directly on the DOM elements.

Here how some other frameworks handle conditional model update:

- yew: the model is mutable  and the update function returns a boolean, to indicate whether a render should be triggered
- ReasonML: `update` returns a new model, like in seed, but if a field in the model is defined as mutable, then an update of only that field won't trigger a `render`. **Would that work in seed ?**

Another approach could be to let `update` return an `Option<Model>` and `None` would avoid a rerender.

[Update: The ReasonML approach will not work, Rust does not allow field mutability, I just learned]

**Comment by rsaccon** [comment]
One more thought on this: My suggestion to let `update` return an `Option<Model>` will cause trouble when calling `update` recursively. 

So maybe `update` could return a tuple like: `(Model, shouldRenderBool)`

Of course this will break everything. But I think, sooner or later, a way to skip render at `update` needs to be provided.

**Comment by David-OConnor** [comment]
Agree - will make breaking/significant changes if needed to get this working.

**Comment by David-OConnor** [comment]
The latest commit has your Tuple idea built in to the lib, and to the counter example. I don't like how it adds verbosity to the API, but I don't have a better approach. Perhaps there's a way to improve this using proc_macros, so that your update func could return just the model for a render, or a (Model, false) tuple for not rendering.

**Comment by rsaccon** [comment]
I also don't like it. Another possibility: returning an enum:
```
enum ModelUpdate {
  ShouldRender(Model),
  SkipRender(Model)
}
```

**Comment by David-OConnor** [comment]
I like that more than Tuple; got it working on my setup. Need to think on this; the enum approach isn't bad, but still adds extra code to every update branch for a minority use case.

**Comment by rsaccon** [comment]
If you go with the enum approach, then maybe shortening the enum name also saves some extra
```
enum Update {
  Render(Model),
  Skip(Model)
}
```

**Comment by David-OConnor** [comment]
Included in v0.2.4 release, using your last suggestion. I think the extra verbosity is worth the flexibility.
